filament/backend/src/opengl/PlatformCocoaGL.mm

```diff
@@ -47,6 +53,77 @@
     void updateOpenGLContext(NSView *nsView, bool resetView, bool clearView);
 };
 
+CocoaGLSwapChain::CocoaGLSwapChain( NSView* inView )
+        : view(inView)
+        , previousBounds(NSZeroRect)
+        , previousWindowFrame(NSZeroRect)
+        , observers([NSMutableArray array])
+        , currentBounds(NSZeroRect)
+        , currentWindowFrame(NSZeroRect) {
+    __weak NSView* weakView = view;
+    
+    void (^noteHandler)(NSNotification *note) = ^(NSNotification *note) {
```

Comment:
This block captures the `this` pointer implicitly. Can you make that more obvious by prefixing the member var access with `this->`?

I think the `SwapChain` object could get destroyed before this notification block executes, in which case we'd crash.

Reply:
I was thinking that the fact that the destructor removes notifications would prevent the notification from running after that, but I put in a weak reference check to the `observers` member to be sure.

But that made me realize that the anonymous block (the one that installs notifications) posted to the main queue also references member variables `observers`, `currentBounds`, and `currentWindowFrame`.  I'll make that explicit and use a weak reference to `observers`.    Doing a check that the weak reference to `observers` is not nil handles the case that the block runs after the `CocoaGLSwapChain` is destroyed.  And then I thought, it's unlikely, but what if the `CocoaGLSwapChain` gets destroyed *while* the block is running, so that some observers get installed after the chance to remove them has passed?  I put in `@synchronized` checks to handle that.

Reply:
Alright, that sounds fine to me.